File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/GameBean.java
package fr.erban.dxitcompanion.game;

import java.io.Serializable;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import fr.erban.dxitcompanion.game.player.PlayerBean;
import fr.erban.dxitcompanion.game.player.TurnScore;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class GameBean implements Serializable {

    private List<PlayerBean> players;

    private int currentTurn;

    private int pointsToWin;

    private boolean finished;

    private String nameWinner;

    public Map<String, Map<String, Integer>> getScoresheet() {
        final Map<String, Map<String, Integer>> scoresheet = new HashMap<>();

        for (final PlayerBean player : players) {

            final List<TurnScore> scorePerTurn = player.getScoresheet();
            final HashMap<String, Integer> scoresheetPlayer = new HashMap<>();
            for (final TurnScore turnScore : scorePerTurn) {
                scoresheetPlayer.put(String.valueOf(turnScore.getTurn()), turnScore.getScore());
            }
            scoresheet.put(player.getName(), scoresheetPlayer);
        }

        return scoresheet;
    }
}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/GameBean.kt
package fr.erban.dxitcompanion.game

import android.os.Parcelable
import fr.erban.dxitcompanion.game.player.PlayerBean
import fr.erban.dxitcompanion.game.player.TurnScore
import kotlinx.parcelize.Parcelize

@Parcelize
data class GameBean(
    val players: List<PlayerBean>,
    val currentTurn: Int = 0,
    val pointsToWin: Int = 0,
    val maxTurns: Int = Int.MAX_VALUE,
    val finished: Boolean = false,
    val nameWinner: String? = null
) : Parcelable {

    fun getScoresheet(): Map<String, Map<String, Int>> =
        players.associate { player ->
            player.name to player.scoresheet.associate { it.turn.toString() to it.score }
        }
}
